core/controllers/questions_list.py

```diff
@@ -63,10 +67,23 @@ def get(self, comma_separated_skill_ids):
 
         (
             question_summaries, merged_question_skill_links,
-            next_start_cursor) = (
-                question_services.get_displayable_question_skill_link_details(
-                    constants.NUM_QUESTIONS_PER_PAGE, skill_ids, start_cursor)
-            )
+            next_offset
+        ) = (
+
+            question_services.get_displayable_question_skill_link_details(
+                constants.NUM_QUESTIONS_PER_PAGE, skill_ids, offset=offset)
+        )
+
+        _, _, temp_next_offset = (
```

Comment:
Why don't you just do a single query with NUM_QUESTIONS_PER_PAGE + 1? Will the result tell you what you need to know?

Reply:
Have changed it. PTAL. Done